crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -243,6 +245,13 @@ impl Instruction {
                 if let Some(constant) = dfg.get_numeric_constant(*value) {
                     let src_typ = dfg.type_of_value(*value);
                     match (typ, src_typ) {
+                        (
+                            Type::Numeric(NumericType::NativeField),
+                            Type::Numeric(NumericType::Unsigned { .. }),
+                        ) => {
+                            // Unsigned -> Field: redefine same constant as Field
+                            SimplifiedTo(dfg.make_constant(constant, typ.clone()))
```

Comment:
Why can't we apply this `t -> Field` optimization for any numeric t?

Reply:
Maybe we can. `Field` -> `Field` sounds fine. I wasn't unsure about `Signed` -> `Field` simply because I've had no exposure to it. I'm not sure how we represent it even. E.g. would `-1` cast to `field_modulus - 1`?

Reply:
> E.g. would -1 cast to field_modulus - 1?

Possibly, with every value over `field_modulus / 2 - 1` being interpreted as negative. @guipublic can you shed some light on the representation of signed integers and their operations here? Is it two's complement in 254 bits?

Reply:
I've added the `Field` -> `Field` case. Seeing as `Signed`s have mixed support throughout the SSA refactor, I figure we can handle the correct cast behaviour as part of another piece of work.